src/v2/api/services/PermissionService.ts

```diff
@@ -48,10 +47,24 @@ export class PermissionService {
     return true;
   }
 
-  checkPermission (roles: DbRole[], permission: string) {
+  async checkParentIdPermission (roleId: string, permission: string): Promise<boolean> {
+    const role = await this.roleRepository.findById(roleId);
+    return this.checkRolePermission(role, permission);
+  }
+
+  async checkRolePermission (role: DbRole, permission: string): Promise<boolean | undefined> {
```

Comment:
In this case, the ternary logic of the other method complicated the current one. A clear example of complicating the subsequent code to process the result.

Reply:
Yes, it's a bit complicated logic, so i have changed the implementation of PermissionService at all :3